### PR TITLE
Staff search endpoint 

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/MTTL-3094)
## Describe this PR

Search staff endpoint 

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test the complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: HousingSearchApi/V1/Controllers/GetStaffListController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        [ProducesResponseType(typeof(APIResponse<BadRequestException>), 400)]
+        [HttpGet, MapToApiVersion("1")]
+        [LogCall(LogLevel.Information)]
+        public async Task<IActionResult> GetStaffList([FromQuery] GetStaffListRequest request)
+        {
+            try
+            {
+                var staffSearchResult = await _getStaffListUseCase.ExecuteAsync(request).ConfigureAwait(false);
+                var apiResponse = new APIResponse<GetStaffListResponse>(staffSearchResult);
+                apiResponse.Total = staffSearchResult.Total();
```
Comment (notatiyyah) [inline HousingSearchApi/V1/Controllers/GetStaffListController.cs:37]: Shouldn't this conversion happen in the usecase?

File: HousingSearchApi/V1/Infrastructure/IndexSelector.cs [hunk 2]
```diff
@@ -33,6 +34,9 @@ public Indices.ManyIndices Create<T>()
             if (type == typeof(QueryableTransaction))
                 return Indices.Index(new List<IndexName> { "transactions" });
 
+            if (type == typeof(QueryableStaff))
+                return Indices.Index(new List<IndexName> { "staffs" });
```
Comment (notatiyyah) [inline HousingSearchApi/V1/Infrastructure/IndexSelector.cs:38]: The index is named 'staff' (staff is plural)

File: HousingSearchApi/V1/Controllers/GetStaffListController.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace HousingSearchApi.V1.Controllers
+{
+    [ApiVersion("1")]
+    [Produces("application/json")]
+    [Route("api/v1/search/staffs")]
```
Comment (notatiyyah) [inline HousingSearchApi/V1/Controllers/GetStaffListController.cs:16]: This should be 'staff' to match the swagger doc

File: HousingSearchApi/V1/Boundary/Responses/GetStaffListResponse.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+using Hackney.Shared.HousingSearch.Domain.Staff;
+using System.Collections.Generic;
+
+namespace HousingSearchApi.V1.Boundary.Responses
+{
+    public class GetStaffListResponse
+    {
+        private long _total;
+
+        public List<Staff> Staffs { get; set; }
```
Comment (notatiyyah) [inline HousingSearchApi/V1/Boundary/Responses/GetStaffListResponse.cs:10]: This should be ''staff' to match the swagger doc

**Review by notatiyyah — COMMENTED** [review]
Looks good, just need to update 'Staffs' to be 'Staff' to match the swagger doc & index name, which has already been set up :)

File: HousingSearchApi/V1/Controllers/GetStaffListController.cs (hunk 1 above)
Comment (humulla) [reply HousingSearchApi/V1/Controllers/GetStaffListController.cs:37]: I don't think so, It's taking place in the controller for the other endpoints as well

**Review by notatiyyah — APPROVED** [review]

**Pull request merged**